### ISSUE TITLE
Tech-debt - obsolete calls - move up minimum_os_version 12? -> 13

### ISSUE THREAD

**Issue by wweevv-johndpope** [TelegramMessenger/Telegram-iOS#829]
I thought I could change the in the Telegram/BUILD file
minimum_os_version = "9.0"

minimum_os_version = "13.0"

but it throws many errors with MTProto
<img width="404" alt="Screen Shot 2022-08-31 at 11 56 23 am" src="https://user-images.githubusercontent.com/109928647/187575708-a9d9e7d5-76df-4806-b975-0fa81121069b.png">

'OSAtomicCompareAndSwapPtr' is deprecated: first deprecated in iOS 10.0 - Use atomic_compare_exchange_strong_explicit(memory_order_relaxed) from <stdatomic.h> instead

I need to use later version because I'm wanting to use supabase - and min version is 13.

UPDATE - WORKing to ios13 - some regression - but this would be helpful for cherry picking. 
Perhaps we could upgrade the SSignal with some concise code to get away from older memory management.
https://github.com/wweevv-johndpope/Telegram-iOS/tree/ios13

this maybe helpful - I managed to leverage (some updated) ReactiveCocoa classes - specifically
https://github.com/Kosoku/Agamotto/blob/97450135385ce4ba966422123265ba7e4e3b2e4c/Agamotto/Private/KAGRACSerialDisposable.m

**MEMORY MANAGEMENT - OBJECTIVE-C upgrades**
to get away from the legacy calls - OSAtomicCompareAndSwapPtr for MTDisposable + SBBlockDisposable
https://github.com/wweevv-johndpope/Telegram-iOS/commit/10b4f465a68152afb0f78d4d341f00c5eb6832e5

that and a bunch of  refactoring to use   os_unfair_lock_lock(&_lock); removes OSSpinLock Unlock etc
https://github.com/wweevv-johndpope/Telegram-iOS/commit/305829a060bcb89f809c8662924b784f6a1a1839

some changes here upgrading things
https://github.com/wweevv-johndpope/Telegram-iOS/commit/e50b489872a27a6402469fb5d4a018880b4bb80b

**Comment by johndpope** [comment]
so I found some library that has been updated for swift signal

https://github.com/johndpope/KTTC-Lite/tree/main/KTTC%20Lite/Service/Packages/SwiftSignalKit/Source

that slightly helps - but tech debt is much larger

MtProtoKit - needs upgrading.

<img width="1380" alt="Screen Shot 2022-08-31 at 12 48 18 pm" src="https://user-images.githubusercontent.com/289994/187581807-b6127b92-52f4-4bf1-91e4-9df156636493.png">

seems to be a much bigger issue 
https://stackoverflow.com/questions/16999749/expected-at-the-end-of-declaration-list

_The problem occurs when Xcode C Language Dialect is set to C99 instead of GNU99_

**Comment by johndpope** [comment]
Given ios 9 came out 2015 (now 7 years old) - I think it should be made obsolete from future builds.
Should be targeting devices 6 years max. 2023 -> ios 11.

**Comment by wweevv-johndpope** [comment]
"-std=gnu99",

//submodules/MTProtoKit/BUILD
```bazel

objc_library(
    name = "MtProtoKit",
    enable_modules = True,
    module_name = "MtProtoKit",
    srcs = glob([
        "Sources/**/*.m",
        "Sources/**/*.h",
    ]),
    copts = [
        "-Werror",
     "-std=gnu99",
    ],
    hdrs = glob([
        "PublicHeaders/**/*.h",
    ]),
    includes = [
        "PublicHeaders",
    ],
    deps = [
        "//submodules/EncryptionProvider:EncryptionProvider",
    ],
    sdk_frameworks = [
        "Foundation",
        "Security",
        "SystemConfiguration",
        "CFNetwork",
    ],
    sdk_dylibs = [
        "libz",
    ],
    visibility = [
        "//visibility:public",
    ],

)
```

**Comment by wweevv-johndpope** [comment]
There's some code rot in stripe SDK too.

<img width="1489" alt="Screen Shot 2022-09-01 at 11 31 45 am" src="https://user-images.githubusercontent.com/109928647/187813414-82d57dee-c387-4135-9534-ec09ba2cb059.png">

**Comment by ali-fareed** [comment]
@wweevv-johndpope I don't think "-std=gnu99" is the answer here.

**Comment by wweevv-johndpope** [comment]
Hi Ali,
I got it all working here - https://github.com/wweevv-johndpope/Telegram-iOS/tree/ios13
didn't need the gun99 -

**Comment by wweevv-johndpope** [comment]
UPDATE - cheatsheet for updating dependencies - 
https://docs.google.com/presentation/d/1ZYwyA_zHjDugt4UH-APJt4qBSt_fbY_286WrgFQvdfA/edit#slide=id.p

**Comment by ali-fareed** [comment]
Fixed in 9.0.1.

**Comment by johndpope** [comment]
is there a time the code launches for v9? I'm in the dark and someone said code is closed source now?
